node/src/components/consensus/era_supervisor.rs

```diff
@@ -711,19 +711,19 @@ where
         })
     }
 
-    pub(super) fn handle_block_added(&mut self, block: Block) -> Effects<Event<I>> {
+    pub(super) fn handle_block_added(
+        &mut self,
+        block_hash: BlockHash,
+        block_header: BlockHeader,
+    ) -> Effects<Event<I>> {
         let our_pk = self.era_supervisor.public_signing_key.clone();
         let our_sk = self.era_supervisor.secret_signing_key.clone();
-        let era_id = block.header().era_id();
-        self.era_supervisor.executed_block(block.header());
+        let era_id = block_header.era_id();
+        self.era_supervisor.executed_block(&block_header);
         let mut effects = if self.era_supervisor.is_validator_in(&our_pk, era_id) {
-            let block_hash = block.hash();
             self.effect_builder
                 .announce_created_finality_signature(FinalitySignature::new(
-                    *block_hash,
-                    era_id,
-                    &our_sk,
-                    our_pk,
+                    block_hash, era_id, &our_sk, our_pk,
```

Comment:
```suggestion
                    block_header.hash(), era_id, &our_sk, our_pk,
```

Reply:
Done in https://github.com/CasperLabs/casper-node/pull/1325/commits/e56eeeefffda3e6d09a9a0d7184240c0e872cb92.